lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -1817,6 +1817,11 @@ static void checkProtocolRefinementRequirements(ProtocolDecl *proto) {
     if (otherProto == proto)
       continue;
 
+    // For any protocol 'P', there is an implied requirement 'Self: Copyable',
+    // unless it was suppressed via `Self: ~Copyable`; so skip if present.
+    if (otherProto->isSpecificProtocol(KnownProtocolKind::Copyable))
```

Comment:
(And if it was suppressed with ~Copyable, falling through this code path will detect the case where something else implies it)